starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+// routes/auth-routes.js
+const express = require("express");
+const authRoutes = express.Router();
+const passport = require('passport');
+
+// --- Passport signup
+// User model
+const User = require("./user");
+
+// Bcrypt to encrypt passwords
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
+
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === "" || password === "") {
+    res.render("auth/signup", { message: "Indicate username and password" });
+    return;
+  }
+
+  User.findOne({ username }, "username", (err, user) => {
+    if (user !== null) {
+      res.render("auth/signup", { message: "The username already exists" });
+      return;
+    }
+
+    const salt = bcrypt.genSaltSync(bcryptSalt);
+    const hashPass = bcrypt.hashSync(password, salt);
+
+    const newUser = new User({
+      username,
+      password: hashPass
+    });
+
+    newUser.save((err) => {
+      if (err) {
+        res.render("auth/signup", { message: "Something went wrong" });
+      } else {
+        res.redirect("/");
+      }
+    });
+  });
+});
+
+
```

Comment:
Here you should your have your LocalStrategy from passport define inside in a route.